src/citrine/resources/dataset.py

```diff
@@ -252,12 +252,22 @@ def register_all(self, data_concepts_resources: List[ResourceType],
 
         """
         resources = list()
-        for resource in (sorted(data_concepts_resources,
-                                key=lambda resource: writable_sort_order(resource.typ))):
-            registered_resource = self.register(resource, dry_run)
-            if isinstance(registered_resource, BaseEntity):
-                resource.uids = registered_resource.uids
-            resources.append(resource)
+        by_type = {}
```

Comment:
You can use a [defaultdict](https://docs.python.org/3.6/library/collections.html#collections.defaultdict) here